src/commands/activity.js

```diff
@@ -67,27 +78,61 @@ function clearPreviousLine () {
   }
 }
 
-function handleEvent (previousEvent, event) {
+function handleEvent (previousEvent, event, handler) {
   if (isTemporaryEvent(previousEvent)) {
-    clearPreviousLine();
+    handler.pop();
   }
 
-  const activityLine = formatActivityLine(event);
-  Logger.println(activityLine);
+  handler.print(event);
 
   return event;
 }
 
-function onEvent (previousEvent, newEvent) {
-  const { event, date, data: { state, action, commit, cause } } = newEvent;
+function onEvent (previousEvent, newEvent, handler) {
+  const { event, date, data: { uuid, state, action, commit, cause } } = newEvent;
   if (event !== 'DEPLOYMENT_ACTION_BEGIN' && event !== 'DEPLOYMENT_ACTION_END') {
     return previousEvent;
   }
-  return handleEvent(previousEvent, { date, state, action, commit, cause, isLast: true });
+  return handleEvent(previousEvent, { date, uuid, state, action, commit, cause, isLast: true }, handler);
+}
+
+function getEventHandler (format, follow) {
+  switch (format) {
+    case 'json': {
+      if (follow) {
+        return {
+          print: (event) => {
+            Logger.println(JSON.stringify(convertEventToJson(event)));
+          },
+          pop: clearPreviousLine,
+          end: () => {},
+        };
+      }
+      else {
+        const buf = [];
+
+        return {
+          print: (event) => buf.push(convertEventToJson(event)),
+          pop: () => buf.pop(),
+          end: () => {
+            Logger.printJson(buf);
+          },
+        };
+      }
+    }
+    case 'human':
+    default: {
+      return {
+        print: (event) => Logger.println(formatActivityLine(event)),
+        pop: clearPreviousLine,
+        end: () => {},
+      };
+    }
+  }
 }
```

Comment:
note: not that easy to read/understand but it's mainly because of how the follow mode works with the erase line etc... Let's go with it but I think we'll need to reconsider the behaviour follow mode in the future.